modules/reliquary/components/RelicLevelUpProgress.tsx

```diff
@@ -0,0 +1,63 @@
+import { Box, BoxProps, Flex, VStack } from '@chakra-ui/layout';
+import { motion } from 'framer-motion';
+import React, { ReactNode, useEffect, useRef, useState } from 'react';
+import { BeetsBox } from '~/components/box/BeetsBox';
+import { relicGetMaturityProgress } from '~/modules/reliquary/lib/reliquary-helpers';
+import useReliquary from '~/modules/reliquary/lib/useReliquary';
+
+interface Props {}
+
+export default function RelicLevelUpProgress({ ...rest }: Props & BoxProps) {
+    const { maturityThresholds = [], selectedRelic } = useReliquary();
+    const { progressToNextLevel } = relicGetMaturityProgress(selectedRelic, maturityThresholds);
+    const progressContainer = useRef<HTMLDivElement>(null);
+    const [progress, setProgress] = useState(0);
+    const levels = maturityThresholds.length;
+
+    useEffect(() => {
+        if (progressContainer.current) {
+            const width = progressContainer.current?.getBoundingClientRect()?.width;
+            const levelWidth = width / levels;
+            const levelProgressWidth = (progressToNextLevel / 100) * levelWidth;
+            const overallProgress = (selectedRelic?.level || 0) / maturityThresholds.length;
+            const x = overallProgress * width + levelProgressWidth;
```

Comment:
I think it should be:
```
            const levelWidth = width / (levels - 1);
            const levelProgressWidth = (progressToNextLevel / 100) * levelWidth;
            const overallProgress = level * levelWidth + levelProgressWidth;
            setProgress(overallProgress / width || 0);
```

Because between the 1 and the 11 there are only 10 levelWidth's needed and not 11.

Also the width does not get set for me on first render on localhost, might this be because of the animation delay? After making a change it does set the width.